### ISSUE TITLE
[Issue] Plugin hyprgrass failed to build again

### ISSUE THREAD

**Issue by JLH42** [horriblename/hyprgrass#179]
**Describe the bug**
It doesn't work after i updated something, and reinstalling it shows that error, i dont remember what i updated so it would be appreciated to tell me what to downgrade or if a fix is possible, thanks

**Log**
```

/tmp/hyprpm/zenbook14 && PKG_CONFIG_PATH="/home/zenbook14/.local/share/hyprpm/headersRoot/share/pkgconfig" meson setup build
The Meson build system
Version: 1.6.0
Source dir: /tmp/hyprpm/zenbook14
Build dir: /tmp/hyprpm/zenbook14/build
Build type: native build
Project name: hyprgrass
Project version: v0.8.2
C compiler for the host machine: cc (gcc 14.2.1 "cc (GCC) 14.2.1 20240910")
C linker for the host machine: cc ld.bfd 2.43.0
C++ compiler for the host machine: c++ (gcc 14.2.1 "c++ (GCC) 14.2.1 20240910")
C++ linker for the host machine: c++ ld.bfd 2.43.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Run-time dependency hyprland found: YES 0.45.2
Found CMake: /usr/bin/cmake (3.31.0)
Run-time dependency wftouch found: NO (tried pkgconfig and cmake)

Executing subproject wf-touch 

wf-touch| Project name: wf-touch
wf-touch| Project version: 0.0
wf-touch| C++ compiler for the host machine: c++ (gcc 14.2.1 "c++ (GCC) 14.2.1 20240910")
wf-touch| C++ linker for the host machine: c++ ld.bfd 2.43.0
wf-touch| Run-time dependency glm found: YES 1.0.1
wf-touch| Run-time dependency doctest found: NO (tried pkgconfig and cmake)
wf-touch| Build targets in project: 1
wf-touch| Subproject wf-touch finished.

Run-time dependency doctest found: NO (tried pkgconfig and cmake)
Configuring version.hpp using configuration
Run-time dependency pixman-1 found: YES 0.44.0
Run-time dependency libinput found: YES 1.26.2
Run-time dependency wayland-server found: YES 1.23.1
Run-time dependency xkbcommon found: YES 1.7.0
Run-time dependency libdrm found: YES 2.4.123
Build targets in project: 3

hyprgrass v0.8.2

  Subprojects
    wf-touch: YES

Found ninja-1.12.1 at /usr/bin/ninja
                                                
 -> cd /tmp/hyprpm/zenbook14 && PKG_CONFIG_PATH="/home/zenbook14/.local/share/hyprpm/headersRoot/share/pkgconfig" ninja -C build
ninja: Entering directory `build'
[1/14] Compiling C++ object src/gestures/libgestures.a.p/Shared.cpp.o
[2/14] Compiling C++ object src/gestures/libgestures.a.p/CompletedGesture.cpp.o
[3/14] Compiling C++ object src/gestures/libgestures.a.p/DragGesture.cpp.o
[4/14] Compiling C++ object subprojects/wf-touch/libwftouch.a.p/src_touch.cpp.o
[5/14] Compiling C++ object subprojects/wf-touch/libwftouch.a.p/src_actions.cpp.o
[6/14] Compiling C++ object src/gestures/libgestures.a.p/Actions.cpp.o
[7/14] Compiling C++ object subprojects/wf-touch/libwftouch.a.p/src_math.cpp.o
[8/14] Linking static target subprojects/wf-touch/libwftouch.a
[9/14] Compiling C++ object src/gestures/libgestures.a.p/Gestures.cpp.o
[10/14] Linking static target src/gestures/libgestures.a
[11/14] Compiling C++ object src/libhyprgrass.so.p/VecSet.cpp.o
../src/VecSet.cpp: In instantiation of ‘bool VecSet<T>::remove(T) [with T = Hyprutils::Memory::CWeakPointer<CWLTouchResource>]’:
../src/VecSet.hpp:27:16:   required from here
   27 | template class VecSet<Hyprutils::Memory::CWeakPointer<CWLTouchResource>>;
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/VecSet.cpp:23:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Hyprutils::Memory::CWeakPointer<CWLTouchResource> >::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   23 |     for (int i = 0; i < this->set.size(); i++) {
      |                     ~~^~~~~~~~~~~~~~~~~~
../src/VecSet.cpp:25:19: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Hyprutils::Memory::CWeakPointer<CWLTouchResource> >::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   25 |             if (i != this->set.size() - 1) {
      |                 ~~^~~~~~~~~~~~~~~~~~~~~~~
[12/14] Compiling C++ object src/libhyprgrass.so.p/GestureManager.cpp.o
FAILED: src/libhyprgrass.so.p/GestureManager.cpp.o 
c++ -Isrc/libhyprgrass.so.p -Isrc -I../src -Isubprojects/wf-touch -I../subprojects/wf-touch -I/usr/include -I/usr/include/pixman-1 -I/usr/include/libdrm -I/home/zenbook14/.local/share/hyprpm/headersRoot/include -I/home/zenbook14/.local/share/hyprpm/headersRoot/include/hyprland/protocols -I/home/zenbook14/.local/share/hyprpm/headersRoot/include/hyprland -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -std=c++23 -fPIC -DWLR_USE_UNSTABLE -MD -MQ src/libhyprgrass.so.p/GestureManager.cpp.o -MF src/libhyprgrass.so.p/GestureManager.cpp.o.d -o src/libhyprgrass.so.p/GestureManager.cpp.o -c ../src/GestureManager.cpp
../src/GestureManager.cpp: In member function ‘bool GestureManager::handleGestureBind(std::string, bool)’:
../src/GestureManager.cpp:204:76: error: ‘class CKeybindManager’ has no member named ‘m_vKeybinds’; did you mean ‘m_lKeybinds’?
  204 |     auto allBinds = std::ranges::views::join(std::array{g_pKeybindManager->m_vKeybinds, this->internalBinds});
      |                                                                            ^~~~~~~~~~~
      |                                                                            m_lKeybinds
[13/14] Compiling C++ object src/libhyprgrass.so.p/main.cpp.o
ninja: build stopped: subcommand failed.


✖ Plugin hyprgrass failed to build.                                                                      
  This likely means that the plugin is either outdated, not yet available for your version, or broken.
  If you are on -git, update first
  Try re-running with -v to see more verbose output.

✔ all plugins built                                                                                      
✔ installed repository                                                                                   
✔ you can now enable the plugin(s) with hyprpm enable                                                    
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Done!    
```

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Arch 6.11.9-arch1-1
 - output of `hyprctl version` or `Hyprland -v`:
   ```
   Hyprland v0.45.2 built against aquamarine 0.5.0
   ```
- output of `hyprctl plugin list`:
```
Plugin hycov by DreamMaoMao:
	Handle: 5e580422b380
	Version: 0.3
	Description: overview mode
```
   ```
   Plugin hyprgrass...
   ```

**Additional context**

**Comment by matt1432** [comment]
This is because you updated hyprgrass without updating Hyprland. You can just update Hyprland and it'll work

**Comment by JLH42** [comment]
![image](https://github.com/user-attachments/assets/b0b740c8-c62f-4715-a097-0b3d70a12fe5)

![image](https://github.com/user-attachments/assets/ce6aa540-869d-47dc-81e5-a30e1c8345ac)

it seems to be the latest so should i reinstall it?

**Comment by matt1432** [comment]
Ah, I mean latest git. You're installing the latest one from arch

**Comment by JLH42** [comment]
thanks it worked,
one question, is the git version generally more compatible and stable? because with the normal version i was having frequent issues with my plugins

**Comment by JLH42** [comment]
FIX : cleanbuild the -git version of hyprland

**Comment by matt1432** [comment]
> is the git version generally more compatible and stable? 

Completely depends on the maintainers of the plugin and how much vaxry makes breaking changes lol

I know that the `hyprpm.toml` has a list of matching commits between a plugin and Hyprland so you can always keep an eye on those

**Comment by horriblename** [comment]
hyprpm pin on v0.45.2 was wrong, fixed in #181
